llarp/service/endpoint.cpp

```diff
@@ -632,9 +673,9 @@ namespace llarp
           if(i == p->intro)
             return;
         }
-        should = true;
+        ++numNotInIntroset;
       });
-      if(m_IntroSet.HasExpiredIntros(now) || should)
+      if(m_IntroSet.HasExpiredIntros(now) || numNotInIntroset > 1)
```

Comment:
to match previous behavior, should be `> 0`.  Presumably this is why a counter is used now rather than a boolean, just double-checking.